### ISSUE TITLE
micronaut-vertx-pg-client throws a ClassNotFoundException: com.ongres.scram.common.stringprep.StringPreparation

### ISSUE THREAD

**Issue by pjgg** [micronaut-projects/micronaut-sql#837]
### Expected Behavior

I would expect to be able to use a postgresDB through a reactive Vertx client as is described in the Micronaut Doc, but I am getting a ClassNotFoundException.

### Actual Behaviour

 Following the official Micronaut [doc](https://micronaut-projects.github.io/micronaut-sql/3.0.x/guide/) about `micronaut-vertx-pg-client` I am getting the following exception:

```
Caused by: java.lang.ClassNotFoundException: com.ongres.scram.common.stringprep.StringPreparation
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
        ... 21 common frames omitted
```
Workaround: Add the following dependency

`implementation('com.ongres.scram:client:2.1')`

This dep is not defined in Micronaut BOM. In my opinion, this dependency should be included as part of  `io.micronaut.sql:micronaut-vertx-pg-client`. 

### Steps To Reproduce

Follow the [documentation,](https://micronaut-projects.github.io/micronaut-sql/3.0.x/guide/) point `9 Configuring Reactive Postgres Client`

For example, you could create a Micronaut app with the following deps:

```
implementation("io.micronaut.sql:micronaut-vertx-pg-client")
    implementation("io.micronaut.rxjava2:micronaut-rxjava2")
    implementation("io.micronaut:micronaut-runtime")
    runtimeOnly("ch.qos.logback:logback-classic")
    implementation("io.micronaut:micronaut-inject")
    implementation ('io.micronaut:micronaut-management')
```

And the following endpoint:

```
package com.example;

import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Produces;
import io.reactivex.Single;

import io.vertx.reactivex.pgclient.PgPool;
import io.vertx.reactivex.sqlclient.Row;
import io.vertx.reactivex.sqlclient.RowIterator;
import jakarta.inject.Inject;

@Controller("/postgres")
public class HelloWorldController {

    @Inject
    PgPool pgClient;

    @Get("/hello")
    @Produces(MediaType.TEXT_PLAIN)
    public String index() {
        return "Hello World";
    }

    @Get
    @Produces(MediaType.TEXT_PLAIN)
    public Single<String> getPostgresVersion() {
        return  pgClient.query("SELECT version()")
                .rxExecute()
                .map(rows -> {
                    RowIterator<Row> rowIt = rows.iterator();
                    if (rowIt.hasNext()) {
                        String version = rowIt.next().getString("version");
                        return version;
                    }

                    return "";
                });
    }
}

```

Launch a Postgres instance

```
docker run --name some-postgres -e POSTGRES_PASSWORD=topsecret -e POSTGRES_USER=user -e POSTGRES_DB=mydb -p 5432:5432 -d postgres:14
```

and then launch your app: `gradle run`

Finally make a query to /postgres

`curl http://localhost:8080/postgres/`

### Environment Information

openjdk 11.0.17 2022-10-18
OpenJDK Runtime Environment Temurin-11.0.17+8 (build 11.0.17+8)
OpenJDK 64-Bit Server VM Temurin-11.0.17+8 (build 11.0.17+8, mixed mode)

Linux fedora 6.1.18-200.fc37.x86_64 

### Version

3.8.7

**Comment by radovanradic** [comment]
Hi @pjgg , have you tried with micronaut 3.9.3?
I have created simple app with your controller and this seems to be working
[pg-vertx-demo.zip](https://github.com/micronaut-projects/micronaut-sql/files/11788012/pg-vertx-demo.zip)
So it may have been fixed or I am missing something.

**Comment by kurosouza** [comment]
I just encountered this error while using Vertx pg client 4.5.1. As described, adding ('com.ongres.scram:client:2.1') dependency fixes it.

**Comment by radovanradic** [comment]
This should be fixed in micronaut-sql 4.5.0.
